modules/core/src/main/java/org/apache/synapse/FaultHandler.java

```diff
@@ -135,4 +143,22 @@ private void traceOrDebugWarn(boolean traceOn, String msg) {
         }
         log.warn(msg);
     }
+
+    /**
+     * Executing the ExtendedSynapseHandler in the error flow.
+     *
+     * @param synCtx Synapse Message Context
+     */
+    protected void executeExtendedSynapseHandler(MessageContext synCtx) {
+        List handlers = synCtx.getEnvironment().getSynapseHandlers();
+        Iterator<SynapseHandler> iterator = handlers.iterator();
+        while (iterator.hasNext()) {
+            SynapseHandler handler = iterator.next();
+            if ((handler instanceof AbstractExtendedSynapseHandler) && (synCtx.getProperty("HAS_COUNTED_ERROR") == null)) {
```

Comment:
Please make this property HAS_COUNTED_ERROR meaningful and add it to constants with comments.

Reply:
Fixed.